docs/contributing.rst

```diff
@@ -354,3 +354,44 @@ In the Cython code, these show up as "preprocessor" defines ``MPI`` and
 
 High-level code can check the version of the HDF5 library, or check to see if
 the method is present on ``FileID`` objects.
+
+
+Testing MPI-only features/code
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Typically to run code under MPI, ``mpirun`` must be used to start the MPI
+processes. Similarly, tests using MPI features (such as collective IO), must
+also be run under ``mpirun``. h5py uses pytest markers (specifically
+``pytest.mark.mpi`` to specify which tests require usage of ``mpirun``, and will
+handle skipping the tests as needed. A simple example of how to do this is::
+
+   @pytest.mark.mpi
+   def test_mpi_feature():
+      import mpi4py
+      # test the MPI feature
+
+To run these tests, you'll need to:
+
+1. Have ``tox`` installed (e.g. via ``pip install tox``)
+2. Have HDF5 built with MPI as per :ref:`build_mpi`
+
+Then running::
+
+   $ CC='mpicc' HDF5_MPI=ON tox -e py37-test-deps-mpi4py
+
+should run the tests. You may need to pass ``HDF5_DIR`` depending on the
+location of the HDF5 with MPI support. You can choose which python version to
+build against by changing py37 (e.g. py27 runs python 2.7, this is a tox
+feature), and test with the minimum version requirements by using ``mindeps``
+rather than ``deps``.
+
+If you get an error similar to::
+
+   There are not enough slots available in the system to satisfy the 4 slots
+   that were requested by the application:
+     python
+
+   Either request fewer slots for your application, or make more slots available
+   for use.
+
+then you need to reduce the number of MPI processes you are asking MPI to use.
```

Comment:
How do you control this? What's the default?

Reply:
I see in tox.ini that the default is 2. I imagine reducing this doesn't make much sense - testing with 1 process is going to miss things. Maybe we should briefly mention how to make more slots available.

Reply:
The OpenMPI flag 

    export OMPI_MCA_rmaps_base_oversubscribe=1

or `mpirun -oversubscribe` might be useful in this context.

Reply:
I've added a section on this mentioning the OMPI env var, I think only OpenMPI has this issue (MPICH apparently doesn't, I've got no idea how to setup MS-MPI and I'm not sure there are any other maintained MPI systems).